### ISSUE TITLE
Accept a closure when adding a type definition

### ISSUE THREAD

**Issue by joente** [thingsdb/ThingsDB#10]
When using `mod_type` to add a new definition to a type ThingsDB requires a fifth argument which to use as default property on each instance of that type.

This fifth value is currently required, but only when having at least one instance.

Consider the following change:
- Do **not** require the fifth argument but let ThingsDB generate a "default" value if none is given
- This default argument should be "as-simple-as-possible", that is:
  - `any` and `nillable` should default to `nil`
  - `int`, `uint` and `number` should default to `0`
  - `pint` and `nint` should default to `1` and `-1`
  - `float` should default to `0.0`  
  - `bool` should default to `false`
  - `str`, `raw`, `utf8` should default to an empty string
  - `bytes` should default to an empty byte string
  - `[..]` and `{..}` should default to an empty array/set
  - `thing` should default to an empty thing
  - `X` should recursively build a default value which is possible because it can never be recursive without finding a `nil` somewhere in the chain.
- Allow a closure as fifth argument with the form: `|self| ...` which will be called on each instance.
  - Argument `self` will be the instance of the type.
  - The new property on `self` should already point to the new value so an instance will not be broken. This should always be done, on all instances. (events should only be generated at this point in code)
  - The return value will be used a new value, except when `nil` is returned in which case we lease the default.
  - ~One warning should be send to the client if a closure is given for which the returning value does not match the required specification~

**Comment by joente** [comment]
Example:
```
mod_type('Channel', 'add', 'chat', 'Chat', |c| {
    // c.chat, and in fact all channels should now point to a default generated object, otherwise the 
    // channel instances would be broken 
    c.chat = Chat{};  // We want each channel to have it's own chat, so create one.
                      // I would like to leave `messages: []` away since this should be default
                      // for an non-nillable array property.
    c.chat.room = `room_{.tmp_chat.id()}`;
    nil;              // ThingsDB will apply the return value unless `nil`. This always is correct
                      // since sometimes you might want to break. If you truly want to set `nil`,
                      // then this still works because each nillable property should default to 
                      // `nil` anyway.
});
```

**Comment by joente** [comment]
We do require a default value for an enumerator as well and we should just take the enum with index 0 as the default. Later on, we can add an action to let the user change the default enum value using a swap internally.

**Comment by joente** [comment]
Resolved in **v0.9.2-alpha-0**
